frontend/src/fixtures/personalSectionsFixtures.js

```diff
@@ -0,0 +1,251 @@
+const personalSectionsFixtures = {
```

Comment:
For these fixtures... which backend API are you relying on?

We need to be sure that the backend API matches the JSON you are using as a basis for this feature.

You don't have to actually hook up the component to the backend yet, but you at least need a "target".   

Is there an existing backend API endpoint that provides the sections in this format, or are you relying on a future one?

Reply:
FYI... the only endpoint that I could find for personal sections is this one:

<img width="516" alt="image" src="https://user-images.githubusercontent.com/1119017/204155256-22f4fde6-6011-4f72-ab58-125c83e311b3.png">

```
GET ​/api​/personalSections​/all  List all sections given a psId
```

And the JSON it returns looks like this:

```
[
  {
    "quarter": "20222",
    "courseId": "CMPTGCS 130H ",
    "title": "ADV TOPICS CMPSC",
    "description": "Advanced topics course for advanced undergraduates,   covering topics not offered in standard computer science   courses, combining research orientation and current   developments and technologies. H. General.",
    "classSections": [
      {
        "enrollCode": "63388",
        "section": "0100",
        "session": null,
        "classClosed": "Y",
        "courseCancelled": null,
        "gradingOptionCode": null,
        "enrolledTotal": 7,
        "maxEnroll": 12,
        "secondaryStatus": null,
        "departmentApprovalRequired": false,
        "instructorApprovalRequired": false,
        "restrictionLevel": null,
        "restrictionMajor": null,
        "restrictionMajorPass": null,
        "restrictionMinor": null,
        "restrictionMinorPass": null,
        "concurrentCourses": [
          "CMPSC   190D 0100"
        ],
        "timeLocations": [
          {
            "room": "143",
            "building": "CRST",
            "roomCapacity": null,
            "days": "M W    ",
            "beginTime": "11:00",
            "endTime": "12:15"
          }
        ],
        "instructors": [
          {
            "instructor": "CONRAD P T",
            "functionCode": "Teaching and in charge"
          }
        ]
      }
    ],
    "generalEducation": [],
    "finalExam": null
  },
  {
    "quarter": "20222",
    "courseId": "CMPSC   100  ",
    "title": "INTRO TEACH METHODS",
    "description": "Designed to train outstanding undergraduates for learning assistant positio ns in CS courses. Lecture/discussion surveys current research and best prac tices in CS pedagogy including student development theories, different peda gogical techniques, and methods for assessing learning. Students gain exper ience working one-on-one with students, fostering positive learning environ ments, and providing feedback on student work. Students who successfully co mplete this course will earn units by serving as an apprentice undergraduat e learning assistant.",
    "classSections": [
      {
        "enrollCode": "60269",
        "section": "0100",
        "session": null,
        "classClosed": "Y",
        "courseCancelled": null,
        "gradingOptionCode": null,
        "enrolledTotal": null,
        "maxEnroll": 25,
        "secondaryStatus": "R",
        "departmentApprovalRequired": false,
        "instructorApprovalRequired": true,
        "restrictionLevel": null,
        "restrictionMajor": null,
        "restrictionMajorPass": null,
        "restrictionMinor": null,
        "restrictionMinorPass": null,
        "concurrentCourses": [],
        "timeLocations": [
          {
            "room": "3526",
            "building": "PHELP",
            "roomCapacity": null,
            "days": "    F  ",
            "beginTime": "12:30",
            "endTime": "13:45"
          }
        ],
        "instructors": [
          {
            "instructor": "CONRAD P T",
            "functionCode": "Teaching and in charge"
          }
        ]
      }
    ],
    "generalEducation": [],
    "finalExam": null
  },
  {
    "quarter": "20222",
    "courseId": "CMPSC   100  ",
    "title": "INTRO TEACH METHODS",
    "description": "Designed to train outstanding undergraduates for learning assistant positio ns in CS courses. Lecture/discussion surveys current research and best prac tices in CS pedagogy including student development theories, different peda gogical techniques, and methods for assessing learning. Students gain exper ience working one-on-one with students, fostering positive learning environ ments, and providing feedback on student work. Students who successfully co mplete this course will earn units by serving as an apprentice undergraduat e learning assistant.",
    "classSections": [
      {
        "enrollCode": "60277",
        "section": "0101",
        "session": null,
        "classClosed": null,
        "courseCancelled": null,
        "gradingOptionCode": null,
        "enrolledTotal": null,
        "maxEnroll": 25,
        "secondaryStatus": null,
        "departmentApprovalRequired": false,
        "instructorApprovalRequired": true,
        "restrictionLevel": null,
        "restrictionMajor": null,
        "restrictionMajorPass": null,
        "restrictionMinor": null,
        "restrictionMinorPass": null,
        "concurrentCourses": [],
        "timeLocations": [
          {
            "room": "3525",
            "building": "PHELP",
            "roomCapacity": null,
            "days": "  W    ",
            "beginTime": "15:00",
            "endTime": "17:50"
          }
        ],
        "instructors": []
      }
    ],
    "generalEducation": [],
    "finalExam": null
  },
  {
    "quarter": "20222",
    "courseId": "CMPSC   156  ",
    "title": "ADV APP PROGRAM",
    "description": "Advanced application programming using a high-level, virtual-machine-based language. Topics include generic programming, exception handling, automatic memory management, and application development, management, and maintenanc e tools, third-party library use, version control, software testing, issue tracking, code review, and working with legacy code.",
    "classSections": [
      {
        "enrollCode": "08250",
        "section": "0100",
        "session": null,
        "classClosed": null,
        "courseCancelled": null,
        "gradingOptionCode": null,
        "enrolledTotal": 71,
        "maxEnroll": 72,
        "secondaryStatus": "R",
        "departmentApprovalRequired": false,
        "instructorApprovalRequired": false,
        "restrictionLevel": null,
        "restrictionMajor": "+CMPSC+CMPEN+CPSCI",
        "restrictionMajorPass": null,
        "restrictionMinor": null,
        "restrictionMinorPass": null,
        "concurrentCourses": [],
        "timeLocations": [
          {
            "room": "1431",
            "building": "SH",
            "roomCapacity": "76",
            "days": " T R   ",
            "beginTime": "14:00",
            "endTime": "15:15"
          }
        ],
        "instructors": [
          {
            "instructor": "CONRAD P T",
            "functionCode": "Teaching and in charge"
          }
        ]
      }
    ],
    "generalEducation": [],
    "finalExam": null
  },
  {
    "quarter": "20222",
    "courseId": "CMPSC   156  ",
    "title": "ADV APP PROGRAM",
    "description": "Advanced application programming using a high-level, virtual-machine-based language. Topics include generic programming, exception handling, automatic memory management, and application development, management, and maintenanc e tools, third-party library use, version control, software testing, issue tracking, code review, and working with legacy code.",
    "classSections": [
      {
        "enrollCode": "08268",
        "section": "0101",
        "session": null,
        "classClosed": "Y",
        "courseCancelled": null,
        "gradingOptionCode": null,
        "enrolledTotal": 24,
        "maxEnroll": 24,
        "secondaryStatus": null,
        "departmentApprovalRequired": false,
        "instructorApprovalRequired": false,
        "restrictionLevel": null,
        "restrictionMajor": "+CMPSC+CMPEN+CPSCI",
        "restrictionMajorPass": null,
        "restrictionMinor": null,
        "restrictionMinorPass": null,
        "concurrentCourses": [],
        "timeLocations": [
          {
            "room": "1431",
            "building": "SH",
            "roomCapacity": "76",
            "days": "  W    ",
            "beginTime": "16:00",
            "endTime": "16:50"
          }
        ],
        "instructors": [
          {
            "instructor": "HEFFERNAN K J",
            "functionCode": "Teaching but not in charge"
          },
          {
            "instructor": "LU A H",
            "functionCode": "Teaching but not in charge"
          }
        ]
      }
    ],
    "generalEducation": [],
    "finalExam": null
  }
]
```

Reply:
As I look at it now, this may be close enough... But it might be a good idea to update the fixtures to match the actual output of that endpoint just to be sure.

When I say "close enough", what I mean is that the fixtures you have are a proper subset of the data returned by the actual JSON endpoint... so that's probably fine.  It's ok for there to be "extra data" as long as there isn't any missing data, or data in the wrong place.

Reply:
oh I see! that makes sense. Yea I was making sure that only data that the personalSectionsTable was using was in the fixture, but I see how its more useful to include all the data even if it isn't being used. Is this something you would like me to fix for this pr?